onnxscript/rewriter/pattern.py

```diff
@@ -1349,15 +1379,19 @@ def try_rewrite(
             # TODO(rama): Remove the opset imports from deleted nodes?
             _update_opset_imports(graph_or_function, replacement_subgraph)
             _update_opset_imports(model.graph, replacement_subgraph)
+            if tracer:
+                tracer.log(self, graph_or_function, node, match, MatchStatus.SUCCESS)
             return replacement_subgraph
+        if tracer:
+            tracer.log(self, graph_or_function, node, match, MatchStatus.NO_MATCH)
         return None
 
     def apply_to_model(
-        self, model: ir.Model, *, commute: bool = False, verbose: int | None = None
+        self, model: ir.Model, *, commute: bool = False, verbose: int | None = None, debug: bool = False
     ):
         # A convenience method to apply the rule to a model. We use a RewriteRuleSet to
         # handle commutative rules.
-        return RewriteRuleSet([self], commute=commute).apply_to_model(model, verbose=verbose)
+        return RewriteRuleSet([self], commute=commute).apply_to_model(model, verbose=verbose, debug=debug)
```

Comment:
Would it be a good idea to call this mode `dry_run`? Or does `debug` do more than that?

Reply:
I called it `traceonly` originally. Thought `debug` would be easier for people ...

Reply:
Anyway, will leave it as such for now. This may be a bit premature to decide. I think allowing users to pass in their own optional "tracer" object (that decides what information to track/log/print) may be better instead of a boolean.